virttest/vt_cluster/proxy.py

```
@@ -0,0 +1,166 @@
+            #        approach would be to use a structured error code mechanism
+            #        instead of dynamically importing and reconstructing exceptions.
+            try:
+                if root_mod_name:
+                    actual_root_mod = importlib.import_module(root_mod_name)
+                    specific_exception_class = getattr(actual_root_mod, exc_type_name)
+                else:
+                    specific_exception_class = getattr(
+                        importlib.import_module("builtins"), exc_type_name, None
+                    )
+                    if specific_exception_class is None:
+                        specific_exception_class = eval(exc_type_name)
```

Ruff (0.13.1)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)